### PR TITLE
✨[amp-story-player] Adds swipe-to-next/previous story

### LINKED ISSUE #26698: [amp-story-player] Add swipe-to-next story
Swiping between stories should be supported

### PR DESCRIPTION
**Pull request by Enriqe**
Adds swiping between stories.

[Demo (mobile only)](https://stamp-demo.firebaseapp.com/examples/amp-story/player.html)

Includes:
* New handlers for `touchstart`, `touchend`, `touchmove` events sent by the AMP doc.
* Tests (plus cleaning previous TODOs)

Does not include:
* RTL support (will come in a follow-up)
* Scroll issue #27184 (swiping on the player will cause a horizontal scroll in the parent page when it has an overflow).
   * Main issue is that the touch event is "swallowed" by the iframe, and there's no way to `preventDefault` or `stopPropagation` on it. This can be verified in this [demo](https://codepen.io/wildlyinaccurate/pen/rzBaBE), notice how the touchEvents are logged outside the iframe but not when they're inside.

Closes #26698
Tracker #26308

### CODE REVIEW

File: css/amp-story-player-iframe.css [hunk 1]
```diff
@@ -41,6 +41,7 @@ main {
 
 .i-amphtml-story-player-loaded iframe {
   opacity: 1;
+  transition: transform 250ms cubic-bezier(0.4, 0, 0.2, 1);
```
Comment (gmajoulet) [inline css/amp-story-player-iframe.css]: How does it feel? It looks a bit slow for a horizontal animation on a phone, but I trust your judgement here :)

File: src/amp-story-player.js [hunk 2]
```diff
@@ -44,6 +44,17 @@ const IframePosition = {
   NEXT: 1,
 };
 
+/**
+ * @enum {number}
+ */
+const SwipingState = {
+  SWIPING_TO_NEXT: 0,
+  SWIPING_TO_PREVIOUS: 1,
+};
```
Comment (gmajoulet) [inline src/amp-story-player.js]: Would `SWIPING_TO_LEFT/RIGHT` make it easier to solve for RTL support? The gesture recognizer would be tied to the content or its direction.

File: src/amp-story-player.js [hunk 3]
```diff
@@ -432,6 +469,187 @@ export class AmpStoryPlayer {
...
+    } else {
+      this.swipingState_ = SwipingState.SWIPING_TO_PREVIOUS;
+      secondaryTranslate = `translate3d(calc(${deltaX}px - 100%), 0, 0)`;
+    }
+
+    const story = this.stories_[this.currentIdx_];
+    const iframe = this.iframes_[story[IFRAME_IDX]];
+    const translate = `translate3d(${deltaX}px, 0, 0)`;
+
+    setStyles(iframe, {
```
Comment (gmajoulet) [inline src/amp-story-player.js]: Did you try to wrap these into requestAnimationFrame to see if that improves performance? It's basically what `mutateElement` would do.

File: test/unit/test-amp-story-player.js [hunk 4]
```diff
@@ -160,19 +177,32 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
...
     const fakeData = {next: true};
-    fireHandler('selectDocument', fakeData);
+    fireHandler['selectDocument']('selectDocument', fakeData);
+
+    const iframes = playerEl.shadowRoot.querySelectorAll('iframe');
+    expect(iframes[0].getAttribute('i-amphtml-iframe-position')).to.eql('-1');
+    expect(iframes[1].getAttribute('i-amphtml-iframe-position')).to.eql('0');
+  });
+
+  it('should navigate when swiping', async () => {
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js]: Can you add a test when swiping on a first (or last) story?

File: css/amp-story-player-iframe.css (hunk 1 above)
Comment (Enriqe) [reply css/amp-story-player-iframe.css]: Cut it to 200ms, it feels better.

File: src/amp-story-player.js (hunk 3 above)
Comment (Enriqe) [reply src/amp-story-player.js]: Done.

File: test/unit/test-amp-story-player.js (hunk 4 above)
Comment (Enriqe) [reply test/unit/test-amp-story-player.js]: Done.

File: src/amp-story-player.js [hunk 5]
```diff
@@ -98,6 +109,17 @@ export class AmpStoryPlayer {
 
     /** @private {number} */
     this.currentIdx_ = 0;
+
+    /** @private {?SwipingState} */
+    this.swipingState_ = null;
```
Comment (gmajoulet) [inline src/amp-story-player.js]: Nit: just to have a complete state machine, `NOT_SWIPING`? We should also set it back to this state on `touchend`

Comment (Enriqe) [reply src/amp-story-player.js]: Done.

**Pull request merged**